sys/vm/vm_kern.c

```diff
@@ -225,6 +225,9 @@ kmem_alloc_attr_domain(int domain, vm_size_t size, int flags, vm_paddr_t low,
 	offset = addr - VM_MIN_KERNEL_ADDRESS;
 	pflags = malloc2vm_flags(flags) | VM_ALLOC_WIRED;
 	prot = (flags & M_EXEC) != 0 ? VM_PROT_ALL : VM_PROT_RW;
+
+	/* XXX: Do we want a M_CAP? */
```

Comment:
Unfortunately I think we might need an `M_NOCAP` if anything.

Reply:
I'd really hope we could make it opt-in vs opt-out, but you are probably right.  I think this also matters more in the purecap kernel where caps will be more ubiquitous.